workflow-templates/phpunit-mysql.yml

```diff
@@ -42,13 +54,13 @@ jobs:
   phpunit-mysql:
     runs-on: ubuntu-latest
 
-    needs: changes
+    needs: [changes, matrix]
     if: needs.changes.outputs.src != 'false'
 
     strategy:
       matrix:
-        php-versions: ['8.0', '8.1', '8.2', '8.3']
-        server-versions: ['master']
+        php-versions: ${{fromJson(needs.matrix.outputs.php-versions)}}
+        server-versions: ${{ fromJson(needs.matrix.outputs.server-max) }}
```

Comment:
Do you want the full `NxM` matrix?

Reply:
Not really, just thinking how this can be used by apps that support multiple versions and currently run something like: https://github.com/nextcloud/quota_warning/blob/main/.github/workflows/phpunit-mysql.yml#L41-L51

Reply:
switched to generating a matrix that covers all php and server versions with not-to-much™ redundancy.

The action itself now also generated the full NxM matrix (trimmed for supported versions) so we can easily switch to that if desired.